dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/file/EditSingleFilePage.java

```diff
@@ -505,6 +346,12 @@ private void populateDatasetUpdateFailureMessage() {
         JH.addMessage(FacesMessage.SEVERITY_FATAL, getBundleString("dataset.message.filesFailure"));
     }
 
+    private void updateEntityWithUpdatedFile() {
+        workingVersion.getFileMetadatas().stream()
+                .filter(fmd -> fmd.getDataFile().getStorageIdentifier().equals(fileMetadata.getDataFile().getStorageIdentifier()))
+                .forEach(fmd -> fmd = fileMetadata);
+    }
```

Comment:
Why this is needed?

Reply:
They are operation on fileMetadata not taken from entity, so they were "merging" it with entity that was supposed to be saved.

Reply:
You mean not attached to datasetVersion?

Reply:
Yea

Reply:
I should probably refractor filemetadata assignment, but at the time I didnt want to dive into it since i was focused about command extraction.